source/initial_temperature/adiabatic_boundary.cc

```diff
@@ -20,65 +20,45 @@
 
 
 #include <aspect/initial_temperature/adiabatic_boundary.h>
-#include <fstream>
-#include <iostream>
+#include <aspect/geometry_model/interface.h>
+#include <aspect/utilities.h>
+#include <deal.II/base/signaling_nan.h>
 
 namespace aspect
 {
   namespace InitialTemperature
   {
-    template <>
-    double
-    AdiabaticBoundary<2>::get_isotherm_depth(const double,
-                                             const double) const
-    {
-      AssertThrow (false, ExcMessage ("The `adiabatic boundary' initial temperature plugin is only implemented for 3d cases."));
-      return 0;
-    }
-
-    template <>
-    double
-    AdiabaticBoundary<3>::get_isotherm_depth(const double latitude,
-                                             const double longitude) const
-    {
-      /**
-       * Loop over the entire array and see if we find a point
-       * that's within delta of what we're looking for. If this
-       * point is not within the delta then test if it is within twice
-       * as much as delta.
-       */
-      for (unsigned int i = 0; i <= depths_iso.size();)
-        {
-          if (std::fabs(latitude - latitudes_iso[i]) <= delta && std::fabs(longitude - longitudes_iso[i]) <= delta)
-            {
-              return depths_iso[i];
-            }
-          else
-            i++;
-        }
-      Assert (false, ExcInternalError());
-      return 0;
-    }
+    template <int dim>
+    AdiabaticBoundary<dim>::AdiabaticBoundary ()
+      :
+      surface_boundary_id(1)
+    {}
 
-    template <>
-    double
-    AdiabaticBoundary<2>::initial_temperature (const Point<2> &) const
+    template <int dim>
+    void
+    AdiabaticBoundary<dim>::initialize ()
     {
-      AssertThrow (false, ExcMessage ("The `adiabatic boundary' initial temperature plugin is only implemented for 3d cases."));
-      return 0;
+      // Find the boundary indicator that represents the surface
+      surface_boundary_id = this->get_geometry_model().translate_symbolic_boundary_name_to_id("outer");
+      std::set<types::boundary_id> surface_boundary_set;
+      surface_boundary_set.insert(surface_boundary_id);
+
+      // The input ascii table contains longitude(radians), colatitude (radians), and LAB depths(m).
```

Comment:
Likely not your fault, but this comment is confusing (did you copy it from somewhere? It might be confusing there as well). I think what it should be saying is the following:
```suggestion
      // The input ascii table contains one data column (LAB depths(m)) in addition to the coordinate columns.
```